### PR TITLE
fix(threads): updating how our main model recieve events

### PR DESCRIPTION
**Pull request by bassrock**
## Summary

Switch home and saves models to recieve events on a thread and only use the main thread when they have what they need.

## PR Checklist:
- [ ] Added Unit / UI tests
- [ ] Self Review (review, clean up, documentation, run tests)
- [ ] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/PocketKit/Services.swift [hunk 1]
```diff
@@ -115,7 +115,7 @@ struct Services {
             userDefaults: userDefaults,
             badgeProvider: UIApplication.shared
         )
-        subscriptionStore = PocketSubscriptionStore(user: user, receiptService: AppStoreReceiptService(client: Services.shared.v3Client))
+        subscriptionStore = PocketSubscriptionStore(user: user, receiptService: AppStoreReceiptService(client: v3Client))
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/Services.swift:118]: @Gio2018 FYI had to change this, there was a circular depedency. Curious how your tests passed.

File: PocketKit/Sources/PocketKit/Services.swift [hunk 2]
```diff
@@ -11,7 +11,7 @@ import SharedPocketKit
 import Kingfisher
 
 struct Services {
-    static let shared = Services()
+    static let shared: Services = { Services() }()
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/Services.swift:14]: lazy loading

Comment (dskuza) [reply PocketKit/Sources/PocketKit/Services.swift:14]: …there's another version of lazy loading other than `lazy var`? I thought this is just using the result of a closure to declare a `let`.

Comment (dskuza) [reply PocketKit/Sources/PocketKit/Services.swift:14]: > Global constants and variables are always computed lazily, in a similar manner to [Lazy Stored Properties](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/properties#Lazy-Stored-Properties). Unlike lazy stored properties, global constants and variables don’t need to be marked with the lazy modifier.

> Local constants and variables are never computed lazily.

TIL. https://docs.swift.org/swift-book/documentation/the-swift-programming-language/properties

**Review by dskuza — APPROVED** [review]

**Pull request merged**